src/app/pages/MarginTradePage/components/ClosedPositionsTable/ClosedPositionRow.tsx

```diff
@@ -1,52 +1,78 @@
-import { AssetRenderer } from 'app/components/AssetRenderer';
 import { TradeProfit } from 'app/components/TradeProfit';
-import { getTradingPositionPrice } from '../../utils/marginUtils';
+import {
+  getClosePositionPrice,
+  getExitTransactionHash,
+  getOpenPositionPrice,
+} from '../../utils/marginUtils';
 import { ActionButton } from 'app/components/Form/ActionButton';
-import { useTranslation } from 'react-i18next';
 import { PositionEventsTable } from '../PositionEventsTable';
-import { translations } from 'locales/i18n';
-import { LoanEvent } from '../OpenPositionsTable/hooks/useMargin_getLoanEvents';
+import { AssetValue } from 'app/components/AssetValue';
+import { toWei } from 'utils/blockchain/math-helpers';
+import {
+  CloseWithSwap,
+  Liquidate,
+  MarginLoansFieldsFragment,
+  Trade,
+} from 'utils/graphql/rsk/generated';
 
 type ClosedPositionRowProps = {
-  items: LoanEvent[];
+  event: MarginLoansFieldsFragment;
 };
 
 export const ClosedPositionRow: React.FC<ClosedPositionRowProps> = ({
-  items,
+  event,
 }) => {
   const { t } = useTranslation();
   const [showDetails, setShowDetails] = useState(false);
-  const closedItem = useMemo(() => items[items.length - 1], [items]);
-  const openedItem = useMemo(() => items[0], [items]);
-  const loanAsset = assetByTokenAddress(closedItem.loanToken);
-  const collateralAsset = assetByTokenAddress(closedItem.collateralToken);
+  const {
+    trade,
+    loanToken: { id: loanTokenId },
+    collateralToken: { id: collateralTokenId },
+    liquidates,
+    closeWithSwaps,
+  } = event;
+  const tradeData = trade as Trade[];
+  const { positionSize, entryLeverage, entryPrice, transaction } = tradeData[0];
+  const loanAsset = assetByTokenAddress(loanTokenId);
+  const collateralAsset = assetByTokenAddress(collateralTokenId);
   const pair = TradingPairDictionary.findPair(loanAsset, collateralAsset);
+
   const position =
     pair?.longAsset === loanAsset
       ? TradingPosition.LONG
       : TradingPosition.SHORT;
-  const leverage = useMemo(() => openedItem.leverage + 1, [
-    openedItem.leverage,
+
+  const leverage = useMemo(() => Number(entryLeverage) + 1, [entryLeverage]);
+  const openPrice = useMemo(() => getOpenPositionPrice(entryPrice, position), [
+    entryPrice,
+    position,
   ]);
-  const openPrice = useMemo(
-    () => getTradingPositionPrice(openedItem, position),
-    [openedItem, position],
-  );
+
   const closePrice = useMemo(
-    () => getTradingPositionPrice(closedItem, position),
-    [closedItem, position],
+    () =>
+      getClosePositionPrice(
+        liquidates as Liquidate[],
+        closeWithSwaps as CloseWithSwap[],
+        position,
+      ),
+    [liquidates, closeWithSwaps, position],
+  );
+
+  const exitTransactionHash = useMemo(
+    () =>
+      getExitTransactionHash(
+        liquidates as Liquidate[],
+        closeWithSwaps as CloseWithSwap[],
+      ),
+    [liquidates, closeWithSwaps],
```

Comment:
Again, by casting `liquidates` and `closeWithSwaps` you're just bypassing Typescript checking, I don't like it. If it can be `undefined`, it's not correct either. The proper way how to handle it is to have some fallback or a different `render` function in case the information are missing.